jormungandr-lib/src/interfaces/block0_configuration/initial_config.rs

```diff
@@ -135,15 +136,28 @@ impl From<BlockchainConfiguration> for ConfigParams {
 
 type StaticStr = &'static str;
 
-custom_error! {pub FromConfigParamsError
-    InitConfigParamMissing { name: StaticStr } = "initial message misses parameter {name}",
-    InitConfigParamDuplicate { name: StaticStr } = "initial message contains duplicate parameter {name}",
-    NumberOfSlotsPerEpoch { source: super::number_of_slots_per_epoch::TryFromNumberOfSlotsPerEpochError } = "Invalid number of slots per epoch",
-    SlotDuration { source: super::slots_duration::TryFromSlotDurationError } = "Invalid slot duration value",
-    ConsensusLeaderId { source: super::leader_id::TryFromConsensusLeaderIdError } = "Invalid consensus leader id",
-    ActiveSlotCoefficient { source: super::active_slot_coefficient::TryFromActiveSlotCoefficientError } = "Invalid active slot coefficient value",
-    KESUpdateSpeed { source: super::kes_update_speed::TryFromKESUpdateSpeedError } = "Invalid KES Update speed value",
-    FeesGoTo { source: super::fees_go_to::TryFromFeesGoToError } = "Invalid FeesGoTo setting",
+#[derive(Debug, Error)]
+pub enum FromConfigParamsError {
+    #[error("initial message misses parameter {name}")]
+    InitConfigParamMissing { name: StaticStr },
+    #[error("initial message contains duplicate parameter {name}")]
+    InitConfigParamDuplicate { name: StaticStr },
+    #[error("Invalid number of slots per epoch")]
+    NumberOfSlotsPerEpoch(
+        #[from] super::number_of_slots_per_epoch::TryFromNumberOfSlotsPerEpochError,
+    ),
+    #[error("Invalid slot duration value")]
+    SlotDuration(#[from] super::slots_duration::TryFromSlotDurationError),
+    #[error("Invalid consensus leader id")]
+    ConsensusLeaderId(#[from] super::leader_id::TryFromConsensusLeaderIdError),
+    #[error("Invalid active slot coefficient value")]
+    ActiveSlotCoefficient(
+        #[from] super::active_slot_coefficient::TryFromActiveSlotCoefficientError,
+    ),
+    #[error("Invalid KES Update speed value")]
+    KESUpdateSpeed(#[from] super::kes_update_speed::TryFromKESUpdateSpeedError),
+    #[error("Invalid FeesGoTo setting")]
+    FeesGoTo(#[from] super::fees_go_to::TryFromFeesGoToError),
```

Comment:
in `custom_error` the field called `source: ` had the `#[source]` implemented. Not having this available will make us lose some info regarding the cause of an error.

Reply:
The documentation on `thiserror` says that

> The `#[from]` attribute always implies that the same field is `#[source]`, so you don't ever need to specify both attributes.

so I believe this should work as you want it to.

Reply:
great, `thiserror` is really the right choice.